### PR TITLE
add `output.ecmaVersion` option

### PR DESCRIPTION
**Pull request by sokra**
* fix import-circle test case, it was incorrect
* add `output.ecmaVersion` options to set emitted code style
* fix a bug with prefetching initial chunks
* InitFragments have `get[End]Content` methods which get a sourceContext
* RuntimeModules have implicit chunk and compilation
* changed default to `output.ecmaVersion: 2015`

**What kind of change does this PR introduce?**
feature

**Did you add tests for your changes?**
existing test

**Does this PR introduce a breaking change?**
yes, default ecma script version increased

**What needs to be documented once your changes are merged?**
* new option `output.ecmaVersion`
* `output.ecmaVersion` defaults to 2015

### CODE REVIEW

File: declarations/WebpackOptions.d.ts [hunk 1]
```diff
@@ -1198,6 +1198,10 @@ export interface OutputOptions {
 	 * Module namespace to use when interpolating filename template string for the sources array in a generated SourceMap. Defaults to `output.library` if not set. It's useful for avoiding runtime collisions in sourcemaps from multiple webpack projects built as libraries.
 	 */
 	devtoolNamespace?: string;
+	/**
+	 * The maximum EcmaScript version of the webpack generated code (doesn't include input source code from modules).
+	 */
+	ecmaVersion?: 5 | number;
```
Comment (evilebottnawi) [inline declarations/WebpackOptions.d.ts]: eslint supports years also, maybe will be good have this ability too?

Comment (sokra) [reply declarations/WebpackOptions.d.ts]: It's years. The schema is `5 | number > 2015`. ES5 has no year, so this is kind of the exception

Comment (evilebottnawi) [reply declarations/WebpackOptions.d.ts]: :+1:

**Comment by evilebottnawi** [comment]
Terser plugin also support output `ecma` https://github.com/terser-js/terser#compress-options i think we need setup this value too

**Comment by sokra** [comment]
> Terser plugin also support output `ecma` https://github.com/terser-js/terser#compress-options i think we need setup this value too

Yep, as separate PR we can default terser ecma to `output.ecmaVersion`

**Pull request merged**